File: packages/openscore-core/src/main/java/io/semantic/openscore/core/services/api/RankingService.java
package io.semantic.openscore.core.services.api;

import io.semantic.openscore.core.api.ApiResponse;
import io.semantic.openscore.core.api.ranking.Ranking;

import javax.ws.rs.*;
import javax.ws.rs.core.MediaType;
import java.util.List;

@Path("ranking")
@Produces(MediaType.APPLICATION_JSON)
@Consumes(MediaType.APPLICATION_JSON)
public interface RankingService {

    @GET
    @Path("/{id}")
    ApiResponse<Ranking> getRanking(@PathParam("id") long id);

    @GET
    @Path("/")
    ApiResponse<List<Ranking>> getAllRanking(@QueryParam("pais") String pais, @QueryParam("size") @DefaultValue("0") int size);


}

File: openscore-ui/services/RankingService.ts
import { ApiResponse } from "../model/ApiResponse";
import { Ranking } from "../model/Ranking";
import rest from "./Rest";

export class RankingService {
  public getAll(pais: string, size: number) {
    const params = { pais: pais, size: size.toString() };
    return rest.get<ApiResponse<Ranking[]>>("ranking", {
      params: params,
    });
  }
}
